gcs-action/src/gcs.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  Storage,
+  UploadResponse,
+  DeleteFileResponse,
+  File
+} from '@google-cloud/storage'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as core from '@actions/core'
+
+const storage = new Storage()
+
+/**
+ * Uploads a file to a bucket
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/uploadFile.js
+ * @param bucketName The name of the bucket
+ * @param filename The file path
+ * @param destination The destination prefix
+ */
+export async function uploadFile(
+  bucketName: string,
+  filename: string,
+  destination?: string
+): Promise<UploadResponse | Error> {
+  interface UploadOptions {
+    gzip: boolean
+    destination?: string
+  }
+  const options: UploadOptions = {gzip: true}
+  if (destination) {
+    options.destination = `${destination}${filename.slice(
+      filename.lastIndexOf('/') + 1
+    )}`
+  }
+  try {
+    const uploadedFile = await storage
+      .bucket(bucketName)
+      .upload(filename, options)
+    console.log(`${filename} uploaded to ${bucketName}.`)
+    return uploadedFile
+  } catch (err) {
+    return err
```

Comment:
It's more common to allow this error to bubble to the main stack instead of returning an error. Just `await storage.` and let the exception throw.

Reply:
Makes sense, I read up a bit more on TS and I think I understand the flow now.